controllers/azurecluster_controller.go

```diff
@@ -167,6 +167,33 @@ func (r *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterSco
 		return reconcile.Result{}, err
 	}
 
+	// Handle backcompat for CidrBlock
+	if clusterScope.Vnet().CidrBlock != "" {
+		message := "vnet cidrBlock is deprecated, use cidrBlocks instead"
+		clusterScope.Info(message)
+		r.Recorder.Eventf(clusterScope.AzureCluster, corev1.EventTypeWarning, "DeprecatedField", message)
+
+		// Set CIDRBlocks if it is not set.
```

Comment:
Should we explicit call out scenario when the `CidrBlock` and `CidrBlocks` are defined `CidrBlocks` is the one used?